code/_core/obj/structure/interactive/teleporter.dm

```diff
@@ -0,0 +1,79 @@
+/obj/structure/interactive/teleporter
+	name = "teleporter"
+	desc = "Beam me up Scotty!"
+	desc_extended = "A machine that is used for sending down or bringing up stuff through the power of nearly instantaneous molecular relaying."
+	icon = 'icons/obj/structure/telesciencepads.dmi'
+	icon_state = "pad-idle"
+	var/teleport_cooldown = 0
+	var/second_cooldown = 60 //how many seconds its on cooldown for
+
+/obj/structure/interactive/teleporter/clicked_on_by_object(mob/caller, atom/object, location, control, params)
+
+	INTERACT_CHECK
+	INTERACT_DELAY(10)
+	SPAM_CHECK(10)
+
+	if(!SSgamemode.active_gamemode.allow_launch)
+		caller.to_chat(span("warning","The teleporter is still setting up, wait until given permission!"))
+		return FALSE
+	if(teleport_cooldown > world.time)
+		caller.to_chat(span("warning","The teleporter is still on cooldown, wait some time before using it again!"))
+		return FALSE
+	if(!length(gps_list))
+		caller.to_chat(span("warning","There are no active GPS currently, try again later!"))
+		return FALSE
+	var/choice1 = input("Send or Receive?", "Teleport Direction") in list("Send", "Receive", "Cancel")|null
+	if(choice1 == "Cancel")
+		caller.to_chat(span("notice","You decided against using the teleporter."))
+		return FALSE
+	var/choice2 = input("Select the GPS for coordinates.", "GPS Selection") as num|null
+	for(var/k2 in gps_list)
```

Comment:
You should make an assoc list where the number (converted to string) is the key and the gps object is the value. It's faster and better for performance this way.